docs/setup-cluster/k8s/internal-task-gateway.rst

```diff
@@ -0,0 +1,128 @@
+
+############################
+ Sample Setup - Development
+############################
+
+For internal testing and development, we provide a simple setup script that uses a dynamic
+provisioner provided by Project Contour.
+
+On a local dev machine, you can use Minikube and Contour as the controller. We provide a script to
+simplify the process. This can be found in `tools/k8s/launch-minikube-with-gateway.sh`.
+
+After you have a working K8s cluster and a Gateway controller running, configure the resource
+manager via master config and start the Determined cluster.
+
+###############
+ Configuration
+###############
+
+Below you'll find details on how to configure your cluster and Determined master to use the Internal
+Task Gateway.
+
+-  Total active proxies will be limited by: maxItems set in the Gateway CRD and the portRange
+   configured for Determined (not exhaustive).
+
+**********************
+ Master Configuration
+**********************
+
+To configure the optional InternalTaskGateway for a K8s resource manager, you need to add a struct
+under `internal_task_gateway` key under each of the desired resource manager configurations.
+
+The config is shown below.
+
+.. code:: yaml
+
+   internal_task_gateway:
+     # GatewayName as defined in the k8s cluster.
+     gateway_name: <GatewayName>
+
+     # GatewayNamespace as defined in the k8s cluster.
+     gateway_namespace: <GatewayNamespace>
+
+     # GatewayIP as defined in the k8s cluster.
+     # GatewayIP can also be a hostname if that is easier for your setup.
+     gateway_ip: <GatewayIP>
+
+     # GWPortStart denotes the inclusive start of the available and exclusive port range to
+     # MLDE for InternalTaskGateway.
+     gateway_port_range_start: <GWPortStart>
+
+     # GWPortEnd denotes the inclusive end of the available and exclusive port range to
+     # MLDE for InternalTaskGateway.
+     gateway_port_range_end: <GWPortEnd>
+
+-  Valid port range starts from 1025 to 65535, inclusive.
+-  GatewayIP is the IP address of the Gateway controller that is visible to the Determined master.
+
+*********
+ Gateway
+*********
+
+In the CRD `gateways.gateway.networking.k8s.io`
+`schema.openAPIV3Schema.properties.spec.properties.listeners.maxItems` defines a max limit of how
+many listeners can be active on a single gateway. This limit sets the upper bound on how many tasks
+can be actively proxied. By default this limit is only 64.
+
+Note that when configuring this number you might hit K8s validation complexity thresholds checks.
+This can be configured and is dependent on each K8s cluster's requirements and setup.
+
+For example to up the number from allowed listeners to 128, you can modify the CRD at the given path
+above and `kubectl apply -f <path-to-crd>`. Make sure to set the value for the version of the spec
+that your Gateway API is going to use.
+
+################
+ Other Dev Docs
+################
+
```

Comment:
####################################
 Running Determined Outside of the K8s Cluster
####################################